### ISSUE TITLE
dotnet restore: Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by h0tbird** [docker/example-voting-app#24]
Something is wrong with `.NET`:

```
log  : Installing runtime.native.System.Net.Http 4.0.1-rc2-24027.
log  : Installing System.Reflection.Emit.ILGeneration 4.0.1-rc2-24027.
log  : Installing System.Reflection.TypeExtensions 4.1.0-rc2-24027.
log  : Installing System.Diagnostics.Tools 4.0.1-rc2-24027.
log  : Installing System.Globalization.Extensions 4.0.1-rc2-24027.
log  : Installing System.Runtime.InteropServices.PInvoke 4.0.0-rc2-24027.
[91mSegmentation fault (core dumped)
[0mThe command '/bin/sh -c dotnet restore -v minimal src/     && dotnet publish -c Release -o ./ src/Worker/     && rm -rf src/ $HOME/.nuget/' returned a non-zero code: 139
Build step 'Execute shell' marked build as failure
Finished: FAILURE
```

**Comment by h0tbird** [comment]
Related to https://github.com/dotnet/dotnet-docker/issues/61

**Comment by hong-duc** [comment]
Hello i have the same issue, have you able to fix this?

**Comment by ManoMarks** [comment]
@hong-duc @h0tbird I'm not able to reproduce this. Can you give me more details about your system set-up etc.?

**Comment by hong-duc** [comment]
@ManoMarks I am using arch linux, I just follow the instruction this page gave me https://github.com/docker/labs/blob/master/beginner/chapters/votingapp.md

**Comment by h0tbird** [comment]
@ManoMarks I am also running on Arch Linux.

```
Linux carbon 4.8.6-1-ARCH #1 SMP PREEMPT Mon Oct 31 18:51:30 CET 2016 x86_64 GNU/Linux
```

My current Docker version is:
```
[0] ~/git/example-voting-app/worker (master|✔)>> docker version         
Client:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.7.3
 Git commit:   6b644ec
 Built:        Thu Oct 27 19:42:59 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.3
 API version:  1.24
 Go version:   go1.7.3
 Git commit:   6b644ec
 Built:        Thu Oct 27 19:42:59 2016
 OS/Arch:      linux/amd64
```

I can reproduce it like this:
```
[0] ~/git/example-voting-app/worker (master|✔)>> docker build --rm -t worker .
Sending build context to Docker daemon 9.728 kB
Step 1 : FROM microsoft/dotnet:1.0.0-preview1
 ---> 5be9548822f7
Step 2 : WORKDIR /app
 ---> Running in 599f8a2bbc61
 ---> 0f3da2c5b4ad
Removing intermediate container 599f8a2bbc61
Step 3 : ADD src/ /app/src/
 ---> 9bfeb83b58f9
Removing intermediate container 8977e9d9398b
Step 4 : RUN dotnet restore -v minimal src/     && dotnet publish -c Release -o ./ src/Worker/     && rm -rf src/ $HOME/.nuget/
 ---> Running in 26949c093ff3
log  : Restoring packages for /app/src/Worker/project.json...
Segmentation fault (core dumped)
The command '/bin/sh -c dotnet restore -v minimal src/     && dotnet publish -c Release -o ./ src/Worker/     && rm -rf src/ $HOME/.nuget/' returned a non-zero code: 139
```

I can fix it like this:
```
[0] ~/git/example-voting-app/worker (master|✚1)>> git diff                     
diff --git a/worker/Dockerfile b/worker/Dockerfile
index 8e9b9b3..47aed18 100644
--- a/worker/Dockerfile
+++ b/worker/Dockerfile
@@ -1,4 +1,4 @@
-FROM microsoft/dotnet:1.0.0-preview1
+FROM microsoft/dotnet:latest
 
 WORKDIR /app
```

```
[0] ~/git/example-voting-app/worker (master|✚1)>> docker build --rm -t worker .
Sending build context to Docker daemon 9.728 kB
Step 1 : FROM microsoft/dotnet:latest
 ---> 4637b51c2438
Step 2 : WORKDIR /app
 ---> Using cache
 ---> 206980059153
Step 3 : ADD src/ /app/src/
 ---> 0ac9d03e2604
Removing intermediate container 7e0c7bcb8479
Step 4 : RUN dotnet restore -v minimal src/     && dotnet publish -c Release -o ./ src/Worker/     && rm -rf src/ $HOME/.nuget/
 ---> Running in 3e7b9a734481
log  : Restoring packages for /app/src/Worker/project.json...
log  : Installing System.Xml.XPath 4.0.1-rc2-24027.
log  : Installing Microsoft.NETCore.Windows.ApiSets 1.0.1-rc2-24027.
log  : Installing System.Xml.XPath.XDocument 4.0.1-rc2-24027.
log  : Installing System.Text.Encoding.CodePages 4.0.1-rc2-24027.
log  : Installing System.Diagnostics.StackTrace 4.0.1-rc2-24027.
log  : Installing System.Xml.XmlDocument 4.0.1-rc2-24027.
log  : Installing System.Diagnostics.FileVersionInfo 4.0.0-rc2-24027.
log  : Installing Microsoft.NETCore.DotNetHost 1.0.1-rc2-002702-00.
log  : Installing Microsoft.NETCore.Runtime.CoreCLR 1.0.2-rc2-24027.
log  : Installing Microsoft.NETCore.Runtime.Native 1.0.2-rc2-24027.
log  : Installing runtime.native.System.IO.Compression 4.1.0-rc2-24027.
log  : Installing Microsoft.NETCore.Targets 1.0.1-rc2-24027.
log  : Installing Microsoft.CodeAnalysis.Common 1.3.0-beta1-20160429-01.
log  : Installing Newtonsoft.Json 9.0.1-beta1.
log  : Installing Microsoft.Win32.Registry 4.0.0-rc2-24027.
log  : Installing System.Collections.Specialized 4.0.1-rc2-24027.
log  : Installing Microsoft.NETCore.DotNetHostResolver 1.0.1-rc2-002702-00.
log  : Installing System.Security.Cryptography.Csp 4.0.0-rc2-24027.
log  : Installing Npgsql 3.1.3.
log  : Installing StackExchange.Redis 1.1.604-alpha.
log  : Installing System.Security.Cryptography.Cng 4.1.0-rc2-24027.
log  : Installing Microsoft.NETCore.App 1.0.0-rc2-3002702.
log  : Installing System.Text.Encoding.Extensions 4.0.11-rc2-24027.
log  : Installing runtime.native.System.Security.Cryptography 4.0.0-rc2-24027.
log  : Installing runtime.native.System.Net.Http 4.0.1-rc2-24027.
log  : Installing System.Net.Http 4.0.1-rc2-24027.
log  : Installing System.Resources.ResourceManager 4.0.1-rc2-24027.
log  : Installing System.Runtime.Serialization.Primitives 4.1.1-rc2-24027.
log  : Installing System.ObjectModel 4.0.12-rc2-24027.
log  : Installing System.Text.RegularExpressions 4.0.12-rc2-24027.
log  : Installing System.IO 4.1.0-rc2-24027.
log  : Installing System.Globalization 4.0.11-rc2-24027.
log  : Installing System.Linq 4.1.0-rc2-24027.
log  : Installing Microsoft.NETCore.Runtime 1.0.2-rc2-24027.
log  : Installing System.IO.Compression 4.1.0-rc2-24027.
log  : Installing System.Collections 4.0.11-rc2-24027.
log  : Installing System.Dynamic.Runtime 4.0.11-rc2-24027.
log  : Installing Microsoft.NETCore.Platforms 1.0.1-rc2-24027.
log  : Installing System.Globalization.Calendars 4.0.1-rc2-24027.
log  : Installing System.Threading 4.0.11-rc2-24027.
log  : Installing System.Threading.Tasks 4.0.11-rc2-24027.
log  : Installing System.AppContext 4.1.0-rc2-24027.
log  : Installing System.Diagnostics.Debug 4.0.11-rc2-24027.
log  : Installing System.IO.Compression.ZipFile 4.0.1-rc2-24027.
log  : Installing System.Reflection.Extensions 4.0.1-rc2-24027.
log  : Installing System.Runtime.Extensions 4.1.0-rc2-24027.
log  : Installing System.Security.Claims 4.0.1-rc2-24027.
log  : Installing System.Text.Encoding 4.0.11-rc2-24027.
log  : Installing System.Xml.XDocument 4.0.11-rc2-24027.
log  : Installing System.Security.Cryptography.OpenSsl 4.0.0-rc2-24027.
log  : Installing System.Reflection 4.1.0-rc2-24027.
log  : Installing System.Security.Principal 4.0.1-rc2-24027.
log  : Installing System.Linq.Expressions 4.0.11-rc2-24027.
log  : Installing runtime.native.System.Net.Security 4.0.1-rc2-24027.
log  : Installing System.Runtime.InteropServices.PInvoke 4.0.0-rc2-24027.
log  : Installing Microsoft.CSharp 4.0.1-rc2-24027.
log  : Installing System.Xml.ReaderWriter 4.0.11-rc2-24027.
log  : Installing System.Threading.Overlapped 4.0.1-rc2-24027.
log  : Installing runtime.native.System 4.0.0-rc2-24027.
log  : Installing System.Security.Principal.Windows 4.0.0-rc2-24027.
log  : Installing System.Diagnostics.Tracing 4.1.0-rc2-24027.
log  : Installing System.Runtime 4.1.0-rc2-24027.
log  : Installing System.Threading.Thread 4.0.0-rc2-24027.
log  : Installing System.Reflection.TypeExtensions 4.1.0-rc2-24027.
log  : Installing System.Runtime.InteropServices.RuntimeInformation 4.0.0-rc2-24027.
log  : Installing Microsoft.Win32.Primitives 4.0.1-rc2-24027.
log  : Installing System.ComponentModel.Primitives 4.0.1-rc2-24027.
log  : Installing System.Threading.Timer 4.0.1-rc2-24027.
log  : Installing System.Diagnostics.Contracts 4.0.1-rc2-24027.
log  : Installing System.Runtime.Handles 4.0.1-rc2-24027.
log  : Installing System.IO.FileSystem.Primitives 4.0.1-rc2-24027.
log  : Installing System.Net.Primitives 4.0.11-rc2-24027.
log  : Installing System.Globalization.Extensions 4.0.1-rc2-24027.
log  : Installing System.Diagnostics.Tools 4.0.1-rc2-24027.
log  : Installing System.Runtime.Numerics 4.0.1-rc2-24027.
log  : Installing System.Reflection.Primitives 4.0.1-rc2-24027.
log  : Installing System.Console 4.0.0-rc2-24027.
log  : Installing System.Net.Sockets 4.1.0-rc2-24027.
log  : Installing System.Data.Common 4.0.1-rc2-24027.
log  : Installing System.Reflection.Emit.ILGeneration 4.0.1-rc2-24027.
log  : Installing System.ComponentModel.TypeConverter 4.0.1-rc2-24027.
log  : Installing Microsoft.CodeAnalysis.VisualBasic 1.3.0-beta1-20160429-01.
log  : Installing System.Collections.Concurrent 4.0.12-rc2-24027.
log  : Installing System.Net.NameResolution 4.0.0-rc2-24027.
log  : Installing System.Net.NetworkInformation 4.1.0-rc2-24027.
log  : Installing System.Runtime.InteropServices 4.1.0-rc2-24027.
log  : Installing System.Net.Security 4.0.0-rc2-24027.
log  : Installing NETStandard.Library 1.5.0-rc2-24027.
log  : Installing System.Threading.ThreadPool 4.0.10-rc2-24027.
log  : Installing System.Security.Cryptography.Algorithms 4.1.0-rc2-24027.
log  : Installing System.Collections.NonGeneric 4.0.1-rc2-24027.
log  : Installing System.Reflection.Emit.Lightweight 4.0.1-rc2-24027.
log  : Installing System.Reflection.Emit 4.0.1-rc2-24027.
log  : Installing System.IO.FileSystem 4.0.1-rc2-24027.
log  : Installing System.Security.Cryptography.X509Certificates 4.1.0-rc2-24027.
log  : Installing Microsoft.NETCore.DotNetHostPolicy 1.0.1-rc2-002702-00.
log  : Installing System.IO.UnmanagedMemoryStream 4.0.1-rc2-24027.
log  : Installing System.Buffers 4.0.0-rc2-24027.
log  : Installing System.Resources.Reader 4.0.0-rc2-24027.
log  : Installing System.Threading.Tasks.Extensions 4.0.0-rc2-24027.
log  : Installing System.ComponentModel 4.0.1-rc2-24027.
log  : Installing System.Diagnostics.DiagnosticSource 4.0.0-rc2-24027.
log  : Installing System.Security.Cryptography.Primitives 4.0.0-rc2-24027.
log  : Installing System.IO.MemoryMappedFiles 4.0.0-rc2-24027.
log  : Installing Microsoft.VisualBasic 10.0.1-rc2-24027.
log  : Installing System.Threading.Tasks.Parallel 4.0.1-rc2-24027.
log  : Installing System.Runtime.Loader 4.0.0-rc2-24027.
log  : Installing System.IO.FileSystem.Watcher 4.0.0-rc2-24027.
log  : Installing System.Threading.Tasks.Dataflow 4.6.0-rc2-24027.
log  : Installing System.Security.Cryptography.Encoding 4.0.0-rc2-24027.
log  : Installing System.Numerics.Vectors 4.1.1-rc2-24027.
log  : Installing System.Reflection.DispatchProxy 4.0.1-rc2-24027.
log  : Installing System.Net.WebHeaderCollection 4.0.1-rc2-24027.
log  : Installing System.Linq.Parallel 4.0.1-rc2-24027.
log  : Installing System.Collections.Immutable 1.2.0-rc2-24027.
log  : Installing System.Net.Requests 4.0.11-rc2-24027.
log  : Installing System.Reflection.Metadata 1.3.0-rc2-24027.
log  : Installing System.Diagnostics.Process 4.1.0-rc2-24027.
log  : Installing System.ComponentModel.Annotations 4.1.0-rc2-24027.
log  : Installing System.Linq.Queryable 4.0.1-rc2-24027.
log  : Installing Microsoft.CodeAnalysis.CSharp 1.3.0-beta1-20160429-01.
log  : Writing lock file to disk. Path: /app/src/Worker/project.lock.json
log  : src/Worker/project.json
log  : Restore completed in 19367ms.
Publishing Worker for .NETCoreApp,Version=v1.0
Project Worker (.NETCoreApp,Version=v1.0) will be compiled because expected outputs are missing
Compiling Worker for .NETCoreApp,Version=v1.0

Compilation succeeded.
    0 Warning(s)
    0 Error(s)

Time elapsed 00:00:01.8192388
 

publish: Published to ./
Published 1/1 projects successfully
 ---> fde8fe3905d2
Removing intermediate container 3e7b9a734481
Step 5 : CMD dotnet Worker.dll
 ---> Running in c51d8ae23204
 ---> 4fa97162b326
Removing intermediate container c51d8ae23204
Successfully built 4fa97162b326
```

**Comment by hong-duc** [comment]
@h0tbird  ahhh!  I am so stupid :smile:

**Comment by walidshaari** [comment]
why is that
? @hong-duc should we use the latest tag for  "microsoft/dotnet" or there is something else?

**Comment by ManoMarks** [comment]
Sounds like something that was fixed in preview2, which is what latest points to

**Comment by ManoMarks** [comment]
fixed in #44
